File: backend/app/clients/mongo.py
import os
from pymongo import MongoClient
from pymongo.database import Database

_client: MongoClient | None = None


def get_db() -> Database:
	global _client
	if _client is None:
		uri = os.getenv("MONGODB_URI")
		if not uri:
			raise RuntimeError("MONGODB_URI is not set.")
		_client = MongoClient(uri)
	return _client["courtly"]

File: frontend/lib/mongo.ts
import { MongoClient, Db } from 'mongodb';

let client: MongoClient | null = null;

export const getDb = async (): Promise<Db> =>
{
	if(!client)
	{
		const uri = process.env.MONGODB_URI;
		if(!uri) throw new Error('MONGODB_URI is not set');
		client = new MongoClient(uri);
		await client.connect();
	}
	return client.db('courtly');
};
